xroot/Overlay.lua

```diff
@@ -116,6 +116,49 @@ local floatingMenu = app.MenuArc()
 local floatingMenuTimer
 local floatingMenuObject
 
+local function getMenuItemsAndDefault()
+  local choices = { "cancel" }
+  local defaultIndex = #choices + 1
+
+  if floatingMenuObject then
+    -- First get the default menu choice
+    local defaultChoice
+    if floatingMenuObject.getFloatingMenuDefaultChoice then
+      defaultChoice = floatingMenuObject:getFloatingMenuDefaultChoice()
+    end
+
+    -- Now get the item list and set the default index
+    if floatingMenuObject.getFloatingMenuItems then
+      local items = floatingMenuObject:getFloatingMenuItems() or {}
+      for i, item in ipairs(items) do
+        local nextIndex = #choices + 1
+        choices[nextIndex] = item
+        if item == defaultChoice then defaultIndex = nextIndex end
+      end
+    end
+  end
+
+  return choices, defaultIndex - 1
```

Comment:
Need to double check this...

Reply:
Fixed. We should only use 0-based at boundaries, otherwise it get's confusing fast.